Documentation/ExtensionArchitecture/HowTo/Localization/Fluid.rst

```diff
@@ -0,0 +1,339 @@
+==============================
+
+Consider you have to translate the following static texts in your Fluid
+template:
+
+.. code-block:: html
+   :caption: EXT:blog_example/Resources/Private/Templates/SomeTemplate.html
+
+   <h3>{post.title}</h3>
+   <p>By: {post.author.fullName}</p>
+   <p>{post.content -> f:format.nl2br()}</p>
+
+   <h3>Comments</h3>
+   <f:for each="{post.comments}" as="comment">
+     {comment.content -> f:format.nl2br()}
+     <hr>
+   </f:for>
+
+To make such texts exchangeable, they have to be removed from the Fluid
+template and inserted into a language file of format :file:`.xlf`. Every text
+fragment which is to be translated is given an identifier (also called key)
+that can be inserted in the Fluid template.
+
+.. index::
+   pair: Fluid; Localization
+   Fluid; f:translate
+
+.. _f-translate:
+
+The translation ViewHelper :html:`f:translate`
+==============================================
+
+To insert translations into a template, Fluid offers the ViewHelper
+:ref:`f:translate <t3viewhelper:typo3-fluid-translate>`.
+
+This ViewHelper has a property called :html:`key` where the identifier of
+the text fragment prefixed by the location file can be provided.
+
+.. code-block:: html
+   :caption: EXT:my_extension/Resources/Private/Templates/SomeTemplate.html
+
+   <f:translate key="LLL:EXT:my_extension/Resources/Private/Language/yourFile.xlf:yourKey" />
+   <!-- or as inline Fluid: -->
+   {f:translate(key: 'LLL:EXT:my_extension/Resources/Private/Language/yourFile.xlf:yourKey')}
+
+
+The text fragment will now be displayed in the current frontend language
+defined in the site configuration if the translation file of the requested
+language can be found in the location of the prefix.
+
+If the key is not found in the translated file or if the language file is not
+found in the language, the key is looked up in the default language file. If
+it is not found there nothing is displayed.
+
+You can provide a default text fragment in the property :html:`default` to
+avoid no text being displayed:
+
+.. code-block:: html
+   :caption: EXT:my_extension/Resources/Private/Templates/SomeTemplate.html
+
+   <f:translate
+       key="LLL:EXT:my_extension/Resources/Private/Language/yourFile.xlf:yourKey"
+       default="No translation available."
+   />
+
+.. _f-translate-extbase:
+
+The translation ViewHelper in Extbase
+=====================================
+
+In Extbase, the translation file can be detected automatically. It is therefore
+possible to omit the language file prefix.
+
+.. code-block:: html
+   :caption: EXT:blog_example/Resources/Private/Templates/SomeTemplate.html
+
+   <f:translate key="comment_header" />
+   <!-- or as inline Fluid: -->
+   {f:translate(key: 'comment_header')}
+
+
+:html:`<f:translate key="comment_header" />` looks up the key in
+:html:`LLL:EXT:blog_example/Resources/Private/Language/locallang.xlf:comment_header`
+**and** overrides the values from :typoscript:`_LOCAL_LANG` Extbase TypoScript.
```

Comment:
As I understand, _LOCAL_LANG can be used for plugins, either Extbase or pibase. We should also link it to the reference, so the user may look up what it is about:
https://docs.typo3.org/m/typo3/reference-typoscript/11.5/en-us/TopLevelObjects/Plugin.html#local-lang-lang-key-label-key

Reply:
The language strings in non-Extbase plugins only get overriden when they are translated via `$this->pi_getLL()` in the ViewHelper they dont get translated as far as I understand the code.